main.py

```
@@ -1,243 +1,625 @@
+    try:
+        for folder in [os.path.dirname(log_file), data_folder, img_folder]:
+            if not os.path.exists(folder):
+                os.makedirs(folder)
+            # Verify write permissions
+            test_file = os.path.join(folder, '.write_test')
+            try:
+                with open(test_file, 'w') as f:
+                    f.write('test')
+                os.remove(test_file)
+            except IOError as e:
+                raise OSError(f"No write permission in {folder}: {e}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)